scripts/usb_scan.py

```diff
@@ -159,7 +158,7 @@ class UsbDevice(UsbObject):
     _PRODUCT_DETAILS = [_VERSION, _ID_VENDOR, _ID_PRODUCT, _BCD_DEVICE, _SERIAL,
                         _CLASS, _CONF_VALUE, _NUM_INTERFACES, _USB_SPEED]
     _PROPS = _PRODUCT_DESC + _PRODUCT_DETAILS
-    _PROPS_NONABLE = _PRODUCT_DESC + [_SERIAL]
+    _PROPS_NONABLE = [*_PRODUCT_DESC, _SERIAL]
```

Comment:
It's not obvious why this transformation is made, nor why it's correct

Reply:
It's also not obvious why any of the transformations in this PR are made.

Both, mail-alarm and usb_scan already support Python3, since this commit in 2019:
https://github.com/xapi-project/xen-api/commit/3179d669739bda1c67e9859a83f4535b18d60476

This commit exactly describes how the changes were made, which is not described here.

The changes are described as:
```py
    remove redundant else after return
    remove redundant parameters in super(), ref https://docs.python.org/3/library/functions.html#super
    use with open() statement
    remove redundant/unused variables
```
Clearly, many changes were made by automated tools, others seem to be manual changes.

Don't mix automated changes with manual changes in a single PR and even a single commit, as this makes them basically needlessly hard to review in a sensible manner.

This raise the chances to overlook such unexplained changes.
Kudos to Pau that he found this change buried in all the automatic reformatting.

While the use "with open()" would have theoretically some merit, if this these scripts were long-running daemons, but they aren't, so any unclosed open files are closed by the OS when the program is terminated.

That is not to say that `with open` should not be used, but just that it is not needed.

Without further explanations, this PR is a code cleanup PR, not a Python3 PR.

From my point of view, this PR is a hard NACK (will not approve), because it:
1. Mixes automated with manual changes, making them hard to review.
1. It is not obviously needed, these tools already work with Python3.

If you like to contribute the `with open` change, open a new PR with only it.

Likewise for other 'else after return', but it is harmless, except that your IDE may warn you about it.

The other changes are even far less of a fixed "problem" that would belong to a Python3 PR.

Reply:
> It's not obvious why this transformation is made, nor why it's correct
The * iterable unpacking operator was introduced in Python 3.5, which is useful as a more readable way of summing iterables into a list, ref https://peps.python.org/pep-0448/

Reply:
The CI tests are failing with a syntax error on this line.

Reply:
> The CI tests are failing with a syntax error on this line.
I will revert this change first, and will do other ticket to remove the python2 CI, since all of existed python scripts covered by unittests are upgraded python3.